File: beecrowd/MenorPilha.java
package beecrowd;

import java.util.*;

public class MenorPilha {
    public static void main(String[] args) {

        Stack<Integer> pilha = new Stack<>();
        Scanner sc = new Scanner(System.in);

        int num = sc.nextInt();
        sc.nextLine(); 

        for (int i = 0; i < num; i++) {
            String str = sc.nextLine(); 

            if (str.equals("POP")) {
                
                if (!pilha.isEmpty()) {
                    pilha.pop();
                }
            } else if (str.equals("MIN")) {
                
                if (!pilha.isEmpty()) {
                    int min = pilha.get(0);
                    for (int j = 0; j < pilha.size(); j++) {
                        if (pilha.get(j) < min) {
                            min = pilha.get(j);
                        }
                    }
                    System.out.println(min);
                }
            } else if (str.startsWith("PUSH")) {

                int valor = Integer.parseInt(str.substring(5));
                pilha.push(valor);
            }
        }
        sc.close();
    }
}

File: beecrowd/MenorPilha.c
#include<stdio.h>
#include<string.h>

int main()
{
    int num;
    scanf("%d",&num);
    int vet[100];
    int n = 0;
    for(int i=0; i<num; i++) {
        char opcao[10];
        scanf("%s",opcao);

        if(strcmp(opcao, "MIN")==0) {
            int min = vet[0];
            for(int i=1; i<n; i++) {
                if(vet[i-1] > vet[i]) {
                    min = vet[i];
                }
            }
            printf("%d\n",min);
        } else if(strcmp(opcao, "POP")==0) { // retirar do final
            n--;
        } else if(strcmp(opcao, "PUSH")==0) { // colocar no final
            int valor;
            scanf("%d",&valor);
            vet[n] = valor;
            n++;
        }
    }

}
